CMakeLists.txt

```diff
@@ -1,27 +1,35 @@
-cmake_minimum_required(VERSION 3.10)
+cmake_minimum_required(VERSION 3.14)
 project(Wasmtime VERSION 0.27.0)
 
-set(CMAKE_CXX_STANDARD 20)
+if (NOT CMAKE_CXX_STANDARD)
+  message(STATUS "Cannot detect C++ Standard. Switching to C++17 by default !!")
+  set(CMAKE_CXX_STANDARD 17)
+endif()
+message(STATUS "CMAKE_CXX_STANDARD is ${CMAKE_CXX_STANDARD}")
+set(WASMTIME_CPP_SUPPORTED_CXX_STANDARDS "17;20")
```

Comment:
Since the standard is presumably increasing at this point could this test to ensure that it's >= 17 in theory?

Reply:
Done